ios/AppLovinMAXAdView.m

```diff
@@ -0,0 +1,155 @@
+    {
+        self.mAdView.customData = customData;
+    }
+}
+
+- (void)setAdaptiveBannerEnabled:(BOOL)adaptiveBannerEnabled
+{
+    self.mAdaptiveBannerEnabled = adaptiveBannerEnabled;
+    
+    if ( self.mAdView )
+    {
+        [self.mAdView setExtraParameterForKey: @"adaptive_banner" value: (adaptiveBannerEnabled ? @"YES" : @"NO")];
+    }
+}
+
+- (void)setAutoRefresh:(BOOL)autoRefresh
+{
+    self.mAutoRefresh = autoRefresh;
+    
+    if ( self.mAdView )
+    {
+        if ( autoRefresh )
+        {
+            [self.mAdView startAutoRefresh];
+        }
+        else
+        {
+            [self.mAdView stopAutoRefresh];
+        }
+    }
+}
+
+- (void)attachAdView
+{
+    // This dispatch_async allows finishing to get all properties
+    dispatch_async(dispatch_get_main_queue(), ^{
+        
+        if ( self.mAdView ) return;
+
+        // If ad unit id and format has been set - create and attach AdView
+        if ( !( [self.mAdUnitId al_isValidString] && self.mAdFormat )) return;
+        
+        self.mAdView = [[MAAdView alloc] initWithAdUnitIdentifier: self.mAdUnitId adFormat: self.mAdFormat sdk: AppLovinMAX.shared.sdk];
+        self.mAdView.frame = (CGRect) { CGPointZero, self.mAdFormat.size };
+        self.mAdView.delegate = AppLovinMAX.shared; // Go through core class for callback forwarding to React Native layer
+        self.mAdView.revenueDelegate = AppLovinMAX.shared;
+        
+        self.mAdView.placement = self.mPlacement;
+        
+        self.mAdView.customData = self.mCustomData;
+        
+        [self.mAdView setExtraParameterForKey: @"adaptive_banner" value: (self.mAdaptiveBannerEnabled ? @"true" : @"false")];
+        
+        // Set this extra parameter to work around a SDK bug that ignores calls to stopAutoRefresh()
+        [self.mAdView setExtraParameterForKey: @"allow_pause_auto_refresh_immediately" value: @"true"];
+        
+        if ( self.mAutoRefresh )
+        {
+            [self.mAdView startAutoRefresh];
+        }
+        else
+        {
+            [self.mAdView stopAutoRefresh];
+        }
+        
+        [self.mAdView loadAd];
+        
+        [self addSubview: self.mAdView];
+        
+        CGSize adViewSize = [self.mAdFormat adaptiveSizeForWidth: CGRectGetWidth(self.frame)];
+        [NSLayoutConstraint activateConstraints: @[[self.mAdView.widthAnchor constraintEqualToConstant: adViewSize.width],
+                                                   [self.mAdView.heightAnchor constraintEqualToConstant: adViewSize.height],
+                                                   [self.mAdView.centerXAnchor constraintEqualToAnchor: self.centerXAnchor],
+                                                   [self.mAdView.centerYAnchor constraintEqualToAnchor: self.centerYAnchor]]];
+        
+    });
+}
+
+// Called when the subview’s superview changes or when the subview is
+// removed from the view hierarchy completely.
+- (void)willRemoveSubview:(UIView *)subview
```

Comment:
What triggers this call? Are you trying to hook into the event when the ad view is unmounted? If so - `- (void)didMoveToWindow` might be the better callback. And you can check against `self.window` being nil or not. If it is nil - it is unmounted.

Also in iOS we don't do explicit nil checks using `==` - we just use `if ( !subview )`.

Also, when overriding superclass callbacks, you need to call `[super ...]` otherwise there wil be undefined behaviour / bugs.

Reply:
Ok, used didMoveToWindow instead.